pkg/cmd/preview/options.go

```diff
@@ -57,6 +59,38 @@ func (o *PreviewOptions) Validate() error {
 	if o.Output != "" && o.Output != jsonOutput {
 		return errors.New("invalid output type, supported types: json")
 	}
+	if err := o.ValidateSpecFile(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *PreviewOptions) ValidateSpecFile() error {
+	if o.SpecFile == "" {
+		return nil
+	}
+	absSF, err := filepath.Abs(o.SpecFile)
+	if err != nil {
+		return err
+	}
+	fi, err := os.Stat(absSF)
+	if err != nil {
+		return err
+	}
+	if fi.IsDir() || !fi.Mode().IsRegular() {
+		return fmt.Errorf("spec file must be a regular file")
+	}
+	absWD, err := filepath.Abs(o.WorkDir)
+	if err != nil {
+		return err
+	}
+	rel, err := filepath.Rel(absWD, absSF)
+	if err != nil {
+		return err
+	}
+	if rel[:3] == ".."+string(filepath.Separator) {
```

Comment:
This validation is too strict. If a Spec file is specified with an absolute path in the current working directory, would it be considered an error?

Reply:
`filepath.Rel` can calculate the relative path between two path, regardless of whether the input path is a relative path or an absolute path. BTW, the relevant usercase has been added.

Reply:
`filepath.Rel` can calculate the relative path between two path, regardless of whether the input path is a relative path or an absolute path. BTW, the relevant usercase has been added.